src/content/en/tools/workbox/guides/advanced-recipes.md

```diff
@@ -53,32 +63,40 @@ if ('serviceWorker' in navigator) {
 ```
 
 This code uses the [`workbox-window`](/web/tools/workbox/modules/workbox-window)
-package to register a service worker and respond if it gets stuck in the waiting
+package to register a service worker and react if it gets stuck in the waiting
 phase.
 
 When a waiting service worker is found we inform the user that an updated
 version of the site is available and prompt them to reload. If they accept the
 prompt, we `postMessage()` the new service worker telling it to run
-`skipWaiting()` meaning it'll start to activate.
+`skipWaiting()`, meaning it'll start to activate. Once the new service worker
+has activated and taken control, we reload the current page, causing the latest
```

Comment:
If activation is blocked for some reason, the new SW will not take control. IIRC that's why I didn't wait for controlling in the previous version fo the recipe.

Reply:
Then what's the point of reloading if activation is blocked?

Are there scenarios in which reloading will clear up whatever is blocking activation?